Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -329,4 +331,98 @@ def planeConformalMatrix {R} [Field R] (a b : R) (hab : a ^ 2 + b ^ 2 ≠ 0) :
 -/
 end Examples
 
+section cardinal
+
+variable (n : ℕ) {𝔽 : Type*} [Field 𝔽] [Fintype 𝔽]
+
+local notation "q" => Fintype.card 𝔽
+
+noncomputable instance {k : ℕ} :
+    Fintype ({ s : Fin k → (Fin n → 𝔽) // LinearIndependent 𝔽 s}) :=
+  Fintype.ofFinite _
+
+noncomputable instance {k : ℕ} {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+    Fintype (Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)) : Set (Fin n → 𝔽)) :=
+  Fintype.ofFinite _
+
+noncomputable instance {k : ℕ} {s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }} :
+    Fintype ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) :=
+  Fintype.ofFinite _
+
+lemma complement_card {k : ℕ} (s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s }):
+    Fintype.card ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) =
+      (q) ^ n - (q) ^ k := by
+  rw [Fintype.card_compl_set, Fintype.card_fun, Fintype.card_fin]
+  simp only [SetLike.coe_sort_coe]
+  rw [card_eq_pow_finrank (K := 𝔽) (V := Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽))),
+    finrank_span_eq_card s.property, Fintype.card_fin]
+
```

Comment:
Since this is a `def` it will need a docstring. So add something like /-- Define equivalence from... -/ in the line above.